src/ethereum_test_types/types.py

```diff
@@ -1248,12 +1248,17 @@ def __init__(
     ):
         """
         Initializes the requests object.
+
+        If `requests_lists` is given as argument, assumes they are sorted by request type
+        ascending, starting from 0.
         """
         if requests_lists is not None:
             assert len(requests) == 0, "requests must be empty if list is provided"
             self.requests_list = []
-            for requests_list in requests_lists:
-                self.requests_list.append(requests_list_to_bytes(requests_list))
+            for type, requests_list in enumerate(requests_lists):
+                self.requests_list.append(
+                    Bytes(bytes([type]) + requests_list_to_bytes(requests_list))
```

Comment:
This breaks EIP-7685 tests, because of this change in the spec: https://github.com/ethereum/EIPs/commit/5ac0297b9ac29255e785ff305feea940db5ac6b4

I think the requests lists from the blocks that are being parsed and passed to this function might be outdated, so the source might need to be fixed first.

Reply:
Ha, I thought this change isn't yet implemented. My change is only about parity between the two ifs in this function here, but still in prague-devnet-4 world.

Reply:
ok, hold on, I think there indeed is another solution - never pass `b''` elements in the `requests_lists`, and then it works as prague-devnet-5. I think I was wrong about that particular piece, some debugging gone wrong on my part.

I can propose an alternative fix in a second

Reply:
OK, take a look now pls. I still ended up adjusting this function instead of source (I tried that too, but then it is problematic to work out the type of the request lists entry).

I might still be deeply confused here, I was certain I knew what was going on, but not anymore :sweat_smile:.